### PR TITLE
[platformvm] record percent connected

### PR DESCRIPTION
**Pull request by patrick-ogrady**
```
# HELP avalanche_P_vm_percent_connected Percent of connected stake
# TYPE avalanche_P_vm_percent_connected gauge
avalanche_P_vm_percent_connected 0.999990373454485
```

### CODE REVIEW

File: vms/platformvm/vm.go [hunk 1]
```diff
@@ -94,6 +94,7 @@ var (
 
 // VM implements the snowman.ChainVM interface
 type VM struct {
+	metrics
```
Comment (patrick-ogrady) [inline vms/platformvm/vm.go:97]: I was copying the `avm` style but it looks like I borked a test: https://github.com/ava-labs/avalanchego/blob/084300fa306c21c177cd699a1e55aa3363d8b56f/vms/avm/vm.go#L69

Comment (patrick-ogrady) [reply vms/platformvm/vm.go:97]:
```
=== RUN   TestUptimeReporting
    vm_test.go:2583: duplicate metrics collector registration attempted
```

File: vms/platformvm/health.go [hunk 2]
```diff
@@ -16,6 +16,9 @@ func (vm *VM) Health() (interface{}, error) {
 	if err != nil {
 		return nil, fmt.Errorf("couldn't get percent connected: %w", err)
 	}
+
+	vm.metrics.percentConnected.Add(percentConnected)
```
Comment (StephenButtolph) [inline vms/platformvm/health.go:20]: Shouldn't this be `.Set` rather than `.Add`?

Comment (patrick-ogrady) [reply vms/platformvm/health.go:20]: Good catch!

**Pull request merged**